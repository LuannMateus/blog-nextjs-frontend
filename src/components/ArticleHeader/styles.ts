import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    max-width: 100%;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
