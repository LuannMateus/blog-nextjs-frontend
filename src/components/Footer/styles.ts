import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    width: 100%;

    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
