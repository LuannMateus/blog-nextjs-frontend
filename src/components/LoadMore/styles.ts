import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    width: 16rem;

    border: 1px solid ${theme.colors.darkText};

    background: ${theme.colors.primary};
    color: ${theme.colors.white};

    display: flex;
    justify-content: center;
    align-items: center;

    padding: ${theme.spacings.small};

    cursor: pointer;

    transition: filter 300ms ease-in-out;

    &:hover {
      filter: brightness(50%);
    }

    &:disabled {
      opacity: 0.4;
    }
  `}
`;
