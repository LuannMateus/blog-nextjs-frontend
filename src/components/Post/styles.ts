import styled, { css } from 'styled-components';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;

    padding-left: ${theme.spacings.small};
  `}
`;
