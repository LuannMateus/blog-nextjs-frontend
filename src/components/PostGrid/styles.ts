import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    max-width: 100%;
    width: 100%;

    margin: 0 auto;

    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacings.large};

    padding: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const NotFound = styled.p`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      padding: 0;
    }
  `}
`;
