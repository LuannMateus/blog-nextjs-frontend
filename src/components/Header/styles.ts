import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    max-width: ${theme.sizes.max};
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    padding-left: ${theme.spacings.small};
    padding-bottom: ${theme.spacings.xhuge};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${HeadingStyles} {
      margin-right: calc(${theme.spacings.small} - 1rem);
    }

    @media ${theme.media.lteMedium} {
      margin: 0;

      flex-direction: column;
    }
  `}
`;

export const Content = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.medium};

    @media ${theme.media.lteMedium} {
      text-align: center;

      margin-top: ${theme.spacings.small};
    }
  `}
`;
