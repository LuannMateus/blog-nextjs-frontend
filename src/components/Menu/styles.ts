import styled, { css } from 'styled-components';
import { Container as LogoLinkStyles } from '../LogoLink/styles';

export const Wrapper = styled.nav`
  ${({ theme }) => css`
    width: 27rem;
    height: 100vh;

    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #000;
    border-radius: 4px;

    padding-top: 1.4rem;

    position: fixed;
    left: -26rem;

    transition: all 300ms ease-in-out;

    z-index: 1000;

    .menuIcon {
      position: relative;
      left: 1.5rem;

      background: #000;
      padding: 0;
    }

    svg {
      width: 3rem;
      color: ${theme.colors.secondary};

      align-self: flex-end;

      padding-right: 1.6rem;

      cursor: pointer;
      transition: filter 300ms ease-in-out;
    }

    svg:hover {
      filter: brightness(50%);
    }

    ${LogoLinkStyles} {
      margin-top: ${theme.spacings.xhuge};
    }

    &.active {
      left: 0rem;
    }
  `}
`;

export const MenuLinkContainer = styled.section`
  ${() => css`
    width: 90%;

    margin-top: 4rem;
  `}
`;
