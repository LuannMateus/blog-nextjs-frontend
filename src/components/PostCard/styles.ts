import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small};
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    img {
      transition: filter 300ms ease-in-out;
      width: 100%;
      max-width: 100%;
    }

    &:hover a {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      filter: brightness(80%);
    }
  `};
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: opacity 300ms ease-in-out;
`;

export const Excerpt = styled.p``;
