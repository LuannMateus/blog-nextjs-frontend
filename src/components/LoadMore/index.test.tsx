import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { LoadMore } from '.';

import props from './mock';

describe('<LoadMore />', () => {
  it('Should render LoadMore button', () => {
    renderTheme(<LoadMore {...props} />);

    expect(screen.getByRole('button')).toBeEnabled();
  });

  it('Should render LoadMore button with disabled prop', () => {
    renderTheme(<LoadMore {...props} isDisabled={true} />);

    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('Should render LoadMore button without isDisabled prop', () => {
    renderTheme(<LoadMore {...props} isDisabled={undefined} />);

    expect(screen.getByRole('button')).toBeEnabled();
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<LoadMore {...props} />);

    expect(container).toMatchSnapshot();
  });
});
