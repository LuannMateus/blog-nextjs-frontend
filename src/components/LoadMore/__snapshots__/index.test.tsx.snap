// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMore /> Should match snapshot 1`] = `
.c0 {
  width: 16rem;
  border: 1px solid #333333;
  background: #000;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.6rem;
  cursor: pointer;
  -webkit-transition: filter 300ms ease-in-out;
  transition: filter 300ms ease-in-out;
}

.c0:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c0:disabled {
  opacity: 0.4;
}

<div>
  <button
    class="c0"
  />
</div>
`;
